test/unit/io/sequence_file/sequence_file_integration_test.cpp

```diff
@@ -135,3 +135,27 @@ TEST(integration, convert_fastq_to_fasta)
     fout.get_stream().flush();
     EXPECT_EQ(reinterpret_cast<std::ostringstream&>(fout.get_stream()).str(), fasta_out);
 }
+
+TEST(integration, tricky_id)
+{
+    // The read/written part of the id should start with '> ' for indicating a sequence.
+    // The sequence id is '>TEST 1'.
+    std::string const input
+    {
+        "> >TEST 1\n"
+        "ACGT\n"
+    };
+
+    std::string const output_comp
+    {
+        "> >TEST 1\n"
+        "ACGT\n"
+    };
+
```

Comment:
I think:

* rename: `output_comp` -> `output`

And either:

* Write `std::string const output = input;`, so one doesn't start looking for the difference between the two strings :)
* Just re-use `input`.

I prefer the former, because it's explicit.